Automattic/woocommerce-payments
# E2E setup tests: Ignore Docker build ./wordpress-xdebug

Fixes #4585 

See p1660267127466569-slack-CGGCLBN58

#### Issue 

Sometimes we see the `wordpress` Docker build process failed like [here](https://github.com/Automattic/woocommerce-payments/runs/7800300481?check_suite_focus=true#step:14:35):

```
===> Starting CLIENT containers
Creating network "env_default" with the default driver
Pulling db (mariadb:10.5.8)...
Building wordpress
The command '/bin/sh -c pecl install xdebug-2.9.8 	&& echo 'xdebug.remote_enable=1' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_port=9000' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_host=host.docker.internal' >> $PHP_INI_DIR/php.ini 	&& echo 'xdebug.remote_autostart=0' >> $PHP_INI_DIR/php.ini 	&& docker-php-ext-enable xdebug > /dev/null' returned a non-zero code: 1
Service 'wordpress' failed to build : Build failed
``` 

The purpose of the Dockerfile build `tests/e2e/env/wordpress-xdebug/Dockerfile` is to add xDebug, some other commands like `subversion mariadb-client less`, and WP-CLI. But we're not using them or they have already been available in the WordPress docker image: 

- We disabled xDebug for CI in the `setup.sh` script after building a new image for it: 
https://github.com/Automattic/woocommerce-payments/blob/2db0e16ddb0d5932888d9f8d7760abd6d2fd4063/tests/e2e/env/setup.sh#L108-L112

- We also use `wordpress:cli` in this `setup.sh` script so no need to embed it in the Dockerfile build:
https://github.com/Automattic/woocommerce-payments/blob/2db0e16ddb0d5932888d9f8d7760abd6d2fd4063/tests/e2e/env/setup.sh#L139-L140


#### Changes proposed in this Pull Request
 
My approach here is pretty aggressive, as I am completely removing the Dockerfile file for build. Though, it might be still useful if we need to eable xDebug for local E2E test sites. If so, we might update this PR to have more flexibility between "a build" for local vs "an image" for CI. 

An added benefit is that without building this Dockerfile, the `Prepare test environment` step is faster between 30-60s. 









#### Testing instructions

- Manually run E2E tests for this branch `e2e/ignore-docker-build-for-client-site` on https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml
- Confirm that all tests run fine.

Here is a run: https://github.com/Automattic/woocommerce-payments/actions/runs/2845444886






-------------------


**Post merge**
